### PR TITLE
Show columns

### PR DESCRIPTION
**Pull request by swissspidy**
Implements https://github.com/wp-cli/ideas/issues/97.

Not sure if it's worth adding to the package, but I figured it might be useful.

### CODE REVIEW

File: src/DB_Command.php [hunk 1]
```diff
@@ -1102,6 +1103,79 @@ public function search( $args, $assoc_args ) {
...
+	 *
+	 * @when after_wp_load
+	 * @alias fields
+	 */
+	public function columns( $args, $assoc_args ) {
+		global $wpdb;
+
+		$format = WP_CLI\Utils\get_flag_value( $assoc_args, 'format' );
+
+		WP_CLI\Utils\wp_get_table_names( [ $args[0] ], [] );
```
Comment (swissspidy) [inline src/DB_Command.php]: Needs to use `array()` syntax.

File: features/db-columns.feature [hunk 2]
```diff
@@ -0,0 +1,40 @@
+Feature: Display information about a given table.
+
+  Scenario: Display information about the wp_posts table
```
Comment (swissspidy) [inline features/db-columns.feature]: Schema might not be the same for older WordPress versions. A minimum version should be added.

File: src/DB_Command.php [hunk 3]
```diff
@@ -1102,6 +1103,79 @@ public function search( $args, $assoc_args ) {
...
+		$columns = $wpdb->get_results(
+			'SHOW COLUMNS FROM ' . $args[0]
+		);
+
+		// Display the rows.
+		$formatter_args = array(
+			'format' => $format,
+		);
+
+		$formatter = new Formatter($formatter_args, array( 'Field', 'Type', 'Null', 'Key', 'Default', 'Extra' ) );
```
Comment (schlessera) [inline src/DB_Command.php]: CS - Space missing before `$formatter_args`.

File: src/DB_Command.php [hunk 4]
```diff
@@ -1102,6 +1103,79 @@ public function search( $args, $assoc_args ) {
...
+	 *     | post_parent           | bigint(20) unsigned | NO   | MUL | 0                   |                |
+	 *     | guid                  | varchar(255)        | NO   |     |                     |                |
+	 *     | menu_order            | int(11)             | NO   |     | 0                   |                |
+	 *     | post_type             | varchar(20)         | NO   | MUL | post                |                |
+	 *     | post_mime_type        | varchar(100)        | NO   |     |                     |                |
+	 *     | comment_count         | bigint(20)          | NO   |     | 0                   |                |
+	 *     +-----------------------+---------------------+------+-----+---------------------+----------------+
+	 *
+	 * @when after_wp_load
+	 * @alias fields
```
Comment (schlessera) [inline src/DB_Command.php]: We should skip the alias and only use multiple names for BC reasons.

File: features/db-columns.feature (hunk 2 above)
Comment (schlessera) [reply features/db-columns.feature]: I think the last change to the database schema was for WP 4.2, so adding the following should work: `@require-wp-4.2`

File: README.md [hunk 5]
```diff
@@ -723,6 +723,40 @@ The size defaults to a human-readable number.
     $ wp db size --size_format=mb
     6
 
+
+
+### wp db columns
+
+Display information about a given table.
```
Comment (schlessera) [inline README.md]: `Display` => `Displays`

File: README.md [hunk 6]
```diff
@@ -723,6 +723,40 @@ The size defaults to a human-readable number.
...
+
+~~~
+wp db columns [<table>] [--format]
+~~~
+
+**OPTIONS**
+
+	[<table>]
+		Table name
```
Comment (schlessera) [inline README.md]: `Table name` => `Name of the database table.`

File: README.md [hunk 7]
```diff
@@ -723,6 +723,40 @@ The size defaults to a human-readable number.
...
+
+	[--format]
+		table, csv, json
```
Comment (schlessera) [inline README.md]: Add description:
```
Render output in a particular format.
```

File: README.md [hunk 8]
```diff
@@ -723,6 +723,40 @@ The size defaults to a human-readable number.
...
+~~~
+wp db columns [<table>] [--format]
+~~~
+
+**OPTIONS**
+
+	[<table>]
+		Table name
+
+	[--format]
```
Comment (schlessera) [inline README.md]: Should be: `[--format=<format>]`

File: README.md [hunk 9]
```diff
@@ -723,6 +723,40 @@ The size defaults to a human-readable number.
...
+		table, csv, json
+		---
+		default: table
+		options:
+		  - table
+		  - csv
+		  - json
```
Comment (schlessera) [inline README.md]: Should include `yaml`

File: src/DB_Command.php (hunk 3 above)
Comment (schlessera) [inline src/DB_Command.php]: Given `$formatter_args` is built in a separate variable, the header fields should get the same treatment, especially as they produce a very long line here.

**Comment by schlessera** [comment]
You should also adapt the file https://github.com/wp-cli/db-command/blob/master/.github/settings.yml to add the `command:db-columns` to the settings.

**Pull request merged**